dist/setup-appliance.sh

```diff
@@ -638,11 +639,44 @@ function prepare_os_settings {
         APACHE_ADDITIONAL_PACKAGES="$HTTPD_SERVICE mod_xforward mod_passenger memcached"
         CONFIGURE_APACHE=0
         OBS_SIGND=signd
+        SIGND_BIN="/usr/sbin/signd"
       ;;
     esac
   done
 }
 
+###############################################################################
+
+function enable_forceprojectkeys {
+  # Only run on initial setup
+  echo "Starting enable_forceprojectkeys"
+  if [ -z "$RUN_INITIAL_SETUP" ];then
+    echo "Not running in intial setup mode. Skipping"
+    return
+  fi
+
+  # This is done manually and not via api to avoid authentication
+  # and service dependency (systemd) problems.
+  cd $apidir
+  echo " - Setting enforce_project_keys in api_production.configurations to '$ENABLE_FORCEPROJECTKEYS'"
+  mysql -u $MYSQL_USER -p$MYSQL_PASS -e "update configurations SET enforce_project_keys=$ENABLE_FORCEPROJECTKEYS" api_production || exit 1
+
+  echo " - Starting 'rails writeconfiguration'"
+  RAILS_ENV=production bin/rails writeconfiguration || exit 1
+}
```

Comment:
This needs running source/repo servers...

Reply:
OTOH I don't even know how the initial writethrough of the configuration works at all...

Reply:
Should be started at this point - setting up database initially also does a `writeconfiguration`